src/SamplesApp/SamplesApp.UITests/MessageDialogTests/MessageDialogTest.cs

```diff
@@ -0,0 +1,30 @@
+﻿using NUnit.Framework;
+using SamplesApp.UITests.TestFramework;
+using Uno.UITest.Helpers;
+
+namespace SamplesApp.UITests.MessageDialogTests
+{
+	[TestFixture]
+	public partial class MessageDialogTest : SampleControlUITestBase
+	{
+		[Test]
+		[AutoRetry]
+		public void When_Click_Outside_Dialog_Expect_No_Dismiss()
+		{
+			Run("UITests.MessageDialog.MessageDialogTest");
+			var button = _app.Marked("showDialogBtn");
+			_app.WaitForElement(button);
+			button.FastTap();
+
+			using var screenshot = TakeScreenshot("BeforeClicking");
+
+			var label = _app.Marked("labelOutside");
+			_app.WaitForElement(label);
+			label.FastTap();
+
+			using var screenshot2 = TakeScreenshot("AfterClicking");
+
+			ImageAssert.AreEqual(screenshot, screenshot2);
```

Comment:
It's a good idea to actually close the dialog after the test is finished, to make sure it doesn't hang around and interfere with other tests.

Reply:
@davidjohnoliver I'm not sure how to do so. Can you also help with my question in https://github.com/unoplatform/uno/pull/6654#issuecomment-892912071? Thanks!

Reply:
Hmm, this is actually a special case since it's a purely native dialog, so it's not obvious how to find the 'close' button inside the dialog (and will differ between platforms).

One approach to try would be:
1. Put eg a CheckBox in the page
2. Use `SetDependencyPropertyValue()` (example: https://github.com/unoplatform/uno/blob/master/src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/TextBoxTests/TextBoxTests.cs#L659) to set the CheckBox to checked from the UI test
3. In the page's code-behind, react to the IsChecked value changing by canceling/closing the IAsyncOperation returned by MessageDialog.ShowAsync()

Does that make sense?

Reply:
It looks like the [test is failing on iOS](https://dev.azure.com/uno-platform/Uno%20Platform/_TestManagement/Runs?runId=1561336&_a=resultSummary&resultId=100000). Can you review the screenshots, it may give some hints into what's missing.

Reply:
@jeromelaban I don't quite understand what's happening for iOS. It looks like the "BeforeClicking" screenshot is taken (and is correct), but the `AfterClicking` screenshot isn't taken. What I do in between is just:

```csharp
			var label = _app.Marked("labelOutside");
			_app.WaitForElement(label);
			label.FastTap();
```

No idea why this doesn't work as expected, and I don't have an iOS to see whether the MessageDialog is light dismissible on iOS. I'll open an issue tracking this for now.